pkg/trace/agent/truncator.go

```diff
@@ -62,6 +69,24 @@ const (
 	MaxMetricsKeyLen = MaxMetaKeyLen
 )
 
+// isStructuredMetaKey returns true when the given key is a structured meta tag.
+// Structured meta tags are identified by their suffixes: .json, .msgpack or
+// .protobuf. Examples include _dd.appsec.json and _dd.iast.json.
+// The suffix also serves describing the serialization format in use.
+func isStructuredMetaKey(key string) bool {
+	var structuredMetaKeySuffixes = [...]string{
+		".json",
+		".msgpack",
+		".protobuf",
+	}
+	for _, suffix := range structuredMetaKeySuffixes {
+		if strings.HasSuffix(key, suffix) {
+
+		}
+	}
+	return false
+}
```

Comment:
Since this is in the hot path, could you benchmark and compare it to a simpler function like
```golang
return strings.HasSuffix(key, ".json") || strings.HasSuffix(key, ". msgpack") || strings.HasSuffix(key, ". protobuf")
```

Reply:
Indeed the loop was useless - I changed it into a fully inlined boolean expression like yours.